### PR TITLE
[HttpFoundation] Removed compatibility layer for PHP <5.4 sessions

### PR DESCRIPTION
**Pull request by afurculita**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?  | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

This is a follow-up of https://github.com/symfony/symfony/pull/24239. This PR removes the compatibility layer added for sessions for PHP <5.4.

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Tests/AppVariableTest.php [hunk 1]
```diff
@@ -47,9 +47,8 @@ public function testEnvironment()
 
     public function testGetSession()
     {
-        $session = $this->getMockBuilder(Session::class)->disableOriginalConstructor()->getMock();
```
Comment (Tobion) [inline src/Symfony/Bridge/Twig/Tests/AppVariableTest.php]: why is this changed again?

Comment (afurculita) [reply src/Symfony/Bridge/Twig/Tests/AppVariableTest.php]: Because of the fact that the session is constructed using an instance of `NativeSessionStorage` that by default used the deprecated `SessionHandlerProxy`, this test did not pass due to the deprecation error. Now there are no deprecations, so no need for disabling the original constructor,

**Review by Tobion — APPROVED** [review]

**Review by chalasr — APPROVED** [review]
CHANGELOG should be updated to mention removals

**Comment by afurculita** [comment]
Changelog updated

**Review by stof — APPROVED** [review]

File: src/Symfony/Bridge/Twig/Tests/AppVariableTest.php (hunk 1 above)
Comment (jakzal) [reply src/Symfony/Bridge/Twig/Tests/AppVariableTest.php]: We should be using `MockArraySessionStorage` rather than `NativeSessionStorage` in tests (or create a mock as before).

**Comment by afurculita** [comment]
AppVeyor failure not related to this PR

**Comment by fabpot** [comment]
Thank you @afurculita.
